sigle/src/modules/analytics/stats/StatsFrame.tsx

```diff
@@ -0,0 +1,178 @@
+  const [data, setData] = useState<StatsData[]>(initialRange);
+
+  const baseUrl = window.location.origin;
+
+  useEffect(() => {
+    fetchWeeklyStats();
+  }, []);
+
+  const fetchWeeklyStats = async () => {
+    setData(initialRange);
+
+    const statsRes = await fetch(
+      `${baseUrl}/api/analytics/historical?dateFrom=${format(
+        weekFromDate,
+        'yyyy-MM-dd'
+      )}&dateGrouping=day`
+    );
+
+    const statsData: AnalyticsHistoricalResponse = await statsRes.json();
+    const stats: StatsData[] = statsData.historical.map((item) => {
+      return {
+        value: item.pageviews,
+        date: item.date,
+        visits: item.visits,
+      };
+    });
+
+    setData(stats);
+  };
+
+  const fetchMonthlyStats = async () => {
+    setData(initialRange);
+
+    const statsRes = await fetch(
+      `${baseUrl}/api/analytics/historical?dateFrom=${format(
+        monthFromDate,
+        'yyyy-MM-dd'
+      )}&dateGrouping=day`
+    );
+
+    const statsData: AnalyticsHistoricalResponse = await statsRes.json();
+    const stats: StatsData[] = statsData.historical.map((item) => {
+      return {
+        value: item.pageviews,
+        date: item.date,
+        visits: item.visits,
+      };
+    });
+    setData(stats);
+  };
+
+  const fetchAllTimeStats = async () => {
+    setData(initialRange);
+
+    const statsRes = await fetch(
+      `${baseUrl}/api/analytics/historical?dateFrom=${FATHOM_MAX_FROM_DATE}&dateGrouping=month`
+    );
+
+    const statsData: AnalyticsHistoricalResponse = await statsRes.json();
+    const stats: StatsData[] = statsData.historical.map((item) => {
+      return {
+        value: item.pageviews,
+        date: item.date,
+        visits: item.visits,
+      };
+    });
+
+    setData(stats);
+  };
+
+  const fetchStats = (value: string) => {
+    switch (value) {
+      case 'weekly':
+        fetchWeeklyStats();
+        break;
+      case 'monthly':
+        fetchMonthlyStats();
+        break;
+      case 'all':
+        fetchAllTimeStats();
+        break;
+      default:
+        fetchWeeklyStats();
+        break;
+    }
+  };
+
+  return (
+    <Box css={{ mb: '$8', position: 'relative' }}>
+      <Flex>
+        <Tabs
+          onValueChange={(value) => fetchStats(value)}
+          css={{ width: '100%' }}
+          defaultValue="weekly"
+        >
+          <TabsList
+            css={{ alignSelf: 'end' }}
+            aria-label="See your total views and visitors"
+          >
+            <TabsTrigger value="weekly">Weekly</TabsTrigger>
+            <TabsTrigger value="monthly">Monthly</TabsTrigger>
+            <TabsTrigger value="all">All</TabsTrigger>
+          </TabsList>
+          <TabsContent value="weekly">
+            <Box
+              css={{
+                mb: '$5',
+                position: 'relative',
+                width: '100%',
+                height: 400,
+              }}
+            >
+              <StatsChart type="week" data={data} />
```

Comment:
As we are using tabs, the StatsChart element is recreated and destroyed every time the type is changing. I am wondering if we were rendering the same view but just changing the props passed down to StatsChart, would the graph change the values with a nice animation?

Reply:
Great idea - updated and this is working much better (no longer a flash of no content between tabs), thanks!